### ISSUE TITLE
GDB is taking up alot of resources

### ISSUE THREAD

**Issue by NoahTheNerd** [GDColon/GDBrowser#215]
GDBrowser takes up so many resources on Colon's servers that it's concerning and affecting other services like RoboTop. I think if we could get some people to investigate this and possibly fix it then it'd be a huge weight lifted from Colon's servers.

**Comment by CheryX** [comment]
Update: It's identified(-ish), generating canvas icons on server-side is taking up a lot of resources.

**Comment by GDasdf** [comment]
by "canvas", do you mean the dependency?
cuz I did try to fix a canvas issue with the dependency, maybe 2.8.0 is taking up resources?
If thats the case, ill just put another pull request

**Comment by sudocode1** [comment]
running canvas client-side rather than server-side would be a good idea to use less resources
the problem is, anything related to icons are served by the api, the places those images are in can be fixed by running canvas client-side (e.g. just getting the assets and then changing the colors), but, the api itself has to have canvas server-side

**Comment by GDColon** [comment]
I'm thinking of moving to client side yeah, I have ideas on how to pull it off. The only issue is the spider/robot code, which needs to be recreated and that SUCKS

**Comment by sudocode1** [comment]
maybe for now keep spider/robots server-side? and everything else client-side, until someone takes up that $50 reward

**Comment by GDColon** [comment]
Good idea probably. And I have a program good idea for figuring out the robot/spider issue

**Comment by CheryX** [comment]
Im trying to make it client side rn. Going to be finished soon

**Comment by GDColon** [comment]
Please leave it to me.

**Comment by nolemretaWxd** [comment]
> I'm thinking of moving to client side yeah

this would also allow to unblock downloading levels since each client will download them on their own

**Comment by GDColon** [comment]
> > I'm thinking of moving to client side yeah
> 
> this would also allow to unblock downloading levels since each client will download them on their own

Everything else will still be on server because of CORS rules. I plan on getting RobTop to whitelist the site though

**Comment by sudocode1** [comment]
is this still an issue? iconkit has now been moved to client side

**Comment by GDColon** [comment]
we good now
